test/unit/options/options_black.cpp

```diff
@@ -0,0 +1,168 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Aina Niemetz, Mudathir Mohamed, Andrew Reynolds
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2021 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * Black box testing of the Solver class of the  C++ API.
+ */
+
+#include <algorithm>
+#include <limits>
+
+#include "options/option_exception.h"
+#include "options/options_public.h"
+#include "test_api.h"
+
+namespace cvc5 {
+
+using namespace api;
+
+namespace test {
+
+class TestBlackOptions : public TestApi
+{
+};
+
+template <class... Ts>
+struct overloaded : Ts...
+{
+  using Ts::operator()...;
+};
+template <class... Ts>
+overloaded(Ts...) -> overloaded<Ts...>;
+
+TEST_F(TestBlackOptions, set)
+{
+  const std::set<std::string> muted{"copyright",
+                                    "help",
+                                    "show-config",
+                                    "show-debug-tags",
+                                    "show-trace-tags",
+                                    "version"};
+  for (const auto& name : options::getNames())
+  {
+    if (muted.count(name))
+    {
+      testing::internal::CaptureStdout();
+    }
+    auto info = d_solver.getOptionInfo(name);
+
+    try
+    {
+      std::visit(
+          overloaded{
+              [this, &name](const OptionInfo::VoidInfo& v) {
```

Comment:
Do we still actually use void options after this PR?

Reply:
Indeed we have two `void` options left: `verbose` and `quiet` (which are really just triggers to modify `verbosity`).
We may want to trade having two additional bools (that are never accessed) in favor of removing the possibility of `void`, I don't have a strong opinion on this.

Reply:
Fewer special cases is probably better. I also don't have a strong opinion on this, but maybe we should get rid of void options entirely.

Reply:
I think I agree, but I'd like to do that in a separate PR.